Game/PowerUps.pde

```diff
@@ -0,0 +1,51 @@
+public class OpponentControlPowerUp extends PowerUp {
+    private int duration = 5; 
+
+    public OpponentControlPowerUp(float w, float h) {
+      super(w, h);
+    }
+
+    @Override
+    public void onCollisionWith(SynchronizedItem item) {
+      if (item instanceof Pacman) {
+        Pacman pacman = (Pacman) item;
+        int currentPlayerId = pacman.getPlayerId();
+        int opponentId = (currentPlayerId == 1) ? 2 : 1;
+        Pacman opponentPacman = findPacmanById(opponentId);
+          
+        if (opponentPacman != null) {
+          takeControl(opponentPacman);
+          Timer controlEndTimer = new OneOffTimer(duration, () -> releaseControl(opponentPacman));
+          page.addTimer(controlEndTimer);
+        }
+        discard();
+      }
+    }
+    
+    @Override
+    public PImage getImage() {
+      return super.getImage();
+    }
+
+    private Pacman findPacmanById(int id) {
+      ArrayList<SynchronizedItem> syncItems= page.getSyncItems();
```

Comment:
Every item has its name. Later I will add a `SynchronizedItem getSyncItem(String name)` method for Page class. So it will be much easier to retrieve a specific item.

Reply:
I have add getSyncItem and change the way of powerups generated and placed to make them could be reused.